### PR TITLE
Add test cases for SocketUrlDetection

### PR DESCRIPTION
**Pull request by fuji-151a**
There is no test class in SocketUrlDetection.
So, I created SocketUrlDetectionTest class and I wrote the test cases.
Please check this pull request.

Thanks.

### CODE REVIEW

File: storm/src/test/storm/mesos/logviewer/SocketUrlDetectionTest.java [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+     */
+    @AfterClass
+    public static void afeterClass() {
+        try {
+            serverSocket.close();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+}
```
Comment (erikdw) [inline storm/src/test/storm/mesos/logviewer/SocketUrlDetectionTest.java]: Please add a newline at the end of the file.  i.e., the final character in the file should be a newline at the end of the `}` line.  I really dislike this behavior of some IDEs and/or editors to not finish lines with a newline...

File: storm/src/test/storm/mesos/logviewer/SocketUrlDetectionTest.java [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+
+public final class SocketUrlDetectionTest {
+
+    /**
+     * Test Target.
+     */
+    private SocketUrlDetection target;
+
+    /**
+     * ServerSocket.
```
Comment (erikdw) [inline storm/src/test/storm/mesos/logviewer/SocketUrlDetectionTest.java]: This comment isn't providing much value, I'd just ditch it.

File: storm/src/test/storm/mesos/logviewer/SocketUrlDetectionTest.java [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+        target = new SocketUrlDetection(expectedPort);
+        int actualPort = target.getPort();
+        assertThat(actualPort, is(expectedPort));
+    }
+
+    /**
+     * Close server socket connection.
+     */
+    @AfterClass
+    public static void afeterClass() {
```
Comment (erikdw) [inline storm/src/test/storm/mesos/logviewer/SocketUrlDetectionTest.java]: typo:  `afeterClass` --> `afterClass`.

File: storm/src/test/storm/mesos/logviewer/SocketUrlDetectionTest.java [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+        boolean actual = target.isReachable();
+        assertTrue(actual);
+    }
+
+    /**
+     * If connection refused, return false.
+     */
+    @Test
+    public void isReachableFalse() {
+        target = new SocketUrlDetection(9999);
```
Comment (erikdw) [inline storm/src/test/storm/mesos/logviewer/SocketUrlDetectionTest.java]: I'd put the magic port number of `9999` into a class-level constant.

Comment (fuji-151a) [inline storm/src/test/storm/mesos/logviewer/SocketUrlDetectionTest.java]: OK, I'll try to use constant to this code.

File: storm/src/test/storm/mesos/logviewer/SocketUrlDetectionTest.java (hunk 3 above)
Comment (fuji-151a) [inline storm/src/test/storm/mesos/logviewer/SocketUrlDetectionTest.java]: Thank you for review.
I'll modify this typo.

File: storm/src/test/storm/mesos/logviewer/SocketUrlDetectionTest.java (hunk 1 above)
Comment (fuji-151a) [inline storm/src/test/storm/mesos/logviewer/SocketUrlDetectionTest.java]: OK. I'll try to rewrite code to put a newline at the end of the file.

File: storm/src/test/storm/mesos/logviewer/SocketUrlDetectionTest.java (hunk 2 above)
Comment (fuji-151a) [inline storm/src/test/storm/mesos/logviewer/SocketUrlDetectionTest.java]: I see. Then I'll delete this comment.

File: storm/src/test/storm/mesos/logviewer/SocketUrlDetectionTest.java [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+        boolean actual = target.isReachable();
+        assertTrue(actual);
+    }
+
+    /**
+     * If connection refused, return false.
+     */
+    @Test
+    public void isReachableFalse() {
+        target = new SocketUrlDetection(PORT);
```
Comment (erikdw) [inline storm/src/test/storm/mesos/logviewer/SocketUrlDetectionTest.java]: Can you add a comment about how this test is assuming that nothing is listening to the chosen port on the machine where the test is run?   i.e., it's totally possible that something could be listening to that port.

Comment (fuji-151a) [inline storm/src/test/storm/mesos/logviewer/SocketUrlDetectionTest.java]: @erikdw OK. I'll try to comment to this test case. 
I have one consultation. I have already used `9999` port in this code. but I think that `9999` port is often used. So, Should I change this port from `9999` to any port which are not used much?

Comment (erikdw) [inline storm/src/test/storm/mesos/logviewer/SocketUrlDetectionTest.java]: @fuji-151a : good question... you'd want it to *not* be an ephemeral port, so I would avoid 32768+.  Also, Mesos Slave/Agent advertises 31000-32000 by default. [1] Maybe we can go with 29999 or something similar?  And let's change the `PORT` constant name to something  like `HOPEFULLY_UNUSED_PORT`?

Reference:
[1] https://unix.stackexchange.com/questions/211647/how-safe-is-it-to-change-the-linux-ephemeral-port-range/237543#237543

Comment (fuji-151a) [inline storm/src/test/storm/mesos/logviewer/SocketUrlDetectionTest.java]: @erikdw Thank you for your answering my questions.
OK, I'll rewrite port number from `9999` to `29999`.
And I'll rewrite constant name from `PORT` to `HOPEFULLY_UNUSED_PORT`

**Comment by fuji-151a** [comment]
@erikdw Thank you for review.
I rewrote code based on your review.
- modified constant name: PORT --> HOPEFULLY_UNUSED_PORT
- chaged port 9999 --> 29999
- added comment to isReachableFalse method

Please check again.

**Comment by erikdw** [comment]
@fuji-151a : thanks for the great comments that make it very clear what is happening.

**Pull request merged**